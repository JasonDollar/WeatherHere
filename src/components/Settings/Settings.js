import React from 'react'

// import Options from '../Options/Options'

import classes from './Settings.module.scss'

const Settings = ({ changeLanguage, showSettingsHandler }) => (
  <div className={classes.Settings}>
    <p onClick={showSettingsHandler}>Done</p>
    <select name="lang" id="lang" onChange={changeLanguage} defaultValue="pl">
      <option value="pl">Polski</option>
      <option value="en">English</option>
    </select>
  </div>
)

export default Settings
