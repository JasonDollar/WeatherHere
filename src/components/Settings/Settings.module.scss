.Settings {
  position: fixed;
  top: 0;
  bottom: 5.4rem;
  left: 0;
  right: 0;
  z-index: 200;
  display: flex;
  flex-direction: column;
  background-color: var(--color-grey-3);
  overflow-y: scroll;
  padding-bottom: 3rem;
  @media only screen and (min-width: 576px) {
    top: calc(var(--screen-height) / 2 - 270px);
    bottom: calc(var(--screen-height) / 2 - 270px);
    left: calc(var(--screen-width) / 2 - 190px);
    right: calc(var(--screen-width) / 2 - 190px);
    overflow: auto;
  }
}

.menu {
  margin: 2rem 0rem;
  
  font-size: var(--big-text-font-size);

  & p {
    margin: 1rem 2rem;
  }


}

.form {
  display: flex;
  flex-direction: column;
  & .radio {
    display: none;
  }
  & .radioButton {
    
    display: inline-block;
    position: absolute;
    top: 1rem;
    right: 2rem;
    opacity: 0;
    & svg {
      width: var(--big-text-font-size);
      height: var(--big-text-font-size);
      transform: scale(1.3);
      fill: var(--color-primary);
    }
  }
  // & .radio:checked ~ & .radioButton {
  //   opacity: 1;
  // }
  & .label {
    position: relative;
    font-size: calc(var(--big-text-font-size) - 4px);
    padding: 1rem 2rem;
    border-top: 1px solid var(--color-border);
    // background-color: white;
    
    &:last-child {
      border-bottom: 1px solid var(--color-border);
    }
  }
  & .label.active {
    background-color: var(--color-grey-2);
  }
  
  // & input {
  //   
  // }
}

// .menuUnits::after {
//   content: "";

//   height: 3rem;
// }