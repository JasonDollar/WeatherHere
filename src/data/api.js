const DARK_API = 'a077c3a2c30623e41ef4a1cafbec622d'
const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_API, DARK_URL }  

