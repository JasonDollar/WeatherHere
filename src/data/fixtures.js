import weatherData from './random'
import languageText from './text-locale' 

export const appTestData = {
  weather: weatherData,
  units: 'si',
  language: 'pl',
  text: languageText,
  timezone: 'Europe/Warsaw',
  locationShortName: 'Plock',
  
}