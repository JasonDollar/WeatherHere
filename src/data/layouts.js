const layouts = {
  pink: {
    id: 'pink',
    name: 'Pink',
    primary: '#dd0055',
    primaryOffset: '#f5689e',
    mobileMenu: '#dd0055',
    antiGraph: '#8884d8',
    secondaryGraph: '#0c1052',
    background: '#fffcfc',
    background2: '#fffcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#eae9e9',
    greyGraph: '#676767',
  },
  pinkDark: {
    id: 'pinkDark',
    name: 'Pink Dark',
    primary: '#dd0055',
    primaryOffset: '#dd0055',
    mobileMenu: '#f5689e',
    antiGraph: '#8884d8',
    secondaryGraph: '#dd0055',
    background: '#272424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#efeee7',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#333',
    greyGraph: '#acacac',
  },
  pumpkin: {
    id: 'pumpkin',
    name: 'Pumpkin Dark',
    primary: '#f48042',
    primaryOffset: '#ef915f',
    mobileMenu: '#f48042',
    antiGraph: '#50c25f',
    secondaryGraph: '#f48042',
    background: '#262424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eee',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  evening: {
    id: 'evening',
    name: 'Evening Dark',
    primary: '#b92b27',
    primaryOffset: '#d84f4b',
    mobileMenu: '#d84f4b',
    antiGraph: '#1565C0',
    secondaryGraph: '#d84f4b',
    background: '#282424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eeeef8',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  sky: {
    id: 'sky',
    name: 'Cool Sky',
    primary: '#2980B9',
    primaryOffset: '#6DD5FA',
    mobileMenu: '#2980B9',
    antiGraph: '#756ef0',
    secondaryGraph: '#2980B9',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7e7ec',
    greyGraph: '#676767',
  },
  harvey: {
    id: 'harvey',
    name: 'Harvey',
    primary: '#2b574b',
    primaryOffset: '#75d1a6',
    mobileMenu: '#386e60',
    antiGraph: '#0daeca',
    secondaryGraph: '#2b574b',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7eae7',
    greyGraph: '#676767',
  },


}

export default layouts



