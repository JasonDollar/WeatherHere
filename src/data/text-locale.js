const pl = {
  header: {
    title: 'Pogoda',
    search: 'Szukaj',
  },  
  layout: {
    search: 'Szukaj',
  },
  current: {
    today: 'Dziś',
    humidity: 'Wilgotność',
    wind: 'Wiatr',
    precipProb: 'Opady',
    precipType: 'Rodzaj opadu',
    precipInt: 'Intensywność opadu',
    appTemp: 'Temperatura odczuwalna',
    dewPoint: 'Punkt rosy',
    pressure: 'Ciśnienie',
    cloudCov: 'Zachmurzenie',
    visibility: 'Widoczność',
    uvIndex: 'Indeks UV',
    sunrise: 'Wschód słońca',
    sunset: 'Zachód słońca',
    moonphase: 'Faza księżyca',
    ozone: 'Ozon',
    windGust: 'Wiatr w porywach',
    location: 'Nieznana lokacja',
  },
  date: {
    weekDay: ['Niedziela', 'Poniedziałek', 'Wtorek', 'Środa', 'Czwartek', 'Piątek', 'Sobota', 'Dziś'],
    weekDayShort: ['Niedz.', 'Pon.', 'Wt.', 'Śr.', 'Czw.', 'Pt.', 'Sob.'],
    today: 'Dziś',
    temperature: 'Temperatura (°C)',
    temperatureMax: 'Temperatura Max.',
    temperatureMin: 'Temperatura Min.',
  },
  hourly: {
    hour: 'Godzina',
    precipProb: 'Możliwość Opadów (%)',
    precipProbShort: 'Możliwość Opadów',
    temperature: 'Temperatura (°C)',
    temperatureShort: 'Temperatura',
  },
  update: 'Ostatnia aktualizacja: ',
}

const en = {
  header: {
    title: 'Weather',
    
  },
  layout: {
    search: 'Search',
  },
  current: {
    today: 'Today',
    humidity: 'Humidity',
    wind: 'Wind',
    precipProb: 'Precipitation chance',
    precipType: 'Precipitation type',
    precipInt: 'Precipitation intensity',
    appTemp: 'Apparent temperature',
    dewPoint: 'Dew point',
    pressure: 'Pressure',
    cloudCov: 'Cloud cover',
    visibility: 'Visibility',
    uvIndex: 'UV Index',
    sunrise: 'Sunrise',
    sunset: 'Sunset',
    moonphase: 'Moon phase',
    ozone: 'Ozone',
    windGust: 'Wind gust',
    location: 'Unknown location',
  },
  date: {
    weekDay: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
    weekDayShort: ['Sun.', 'Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.'],
    today: 'Today',
    temperature: 'Temperature (°C)',
    temperatureMax: 'Temperature Max.',
    temperatureMin: 'Temperature Min',
  },
  hourly: {
    hour: 'Hour',
    precipProb: 'Precipitation chance (%)',
    precipProbShort: 'Precipitation chance',
    temperature: 'Temperature (°C)',
    temperatureShort: 'Temperature',
  },
  update: 'Last updated: ',
}

export { pl, en } 