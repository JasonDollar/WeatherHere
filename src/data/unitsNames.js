export const unitsNames = {
  si: {
    id: 'si',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'm/s',
    pressure: 'hPa',
  },
  us: {
    id: 'us',
    distance: 'mi',
    precip: 'in/h',
    precipAcc: 'in',
    temperature: '째F',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  uk2: {
    id: 'uk2',
    distance: 'mi',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  ca: {
    id: 'ca',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'km/h',
    pressure: 'hPa',
  },
}