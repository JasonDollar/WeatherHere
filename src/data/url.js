const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_URL }