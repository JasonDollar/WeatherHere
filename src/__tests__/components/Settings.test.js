import React from 'react'
import { shallow } from 'enzyme'
import { appTestData as data } from '../../data/fixtures'
import Settings from '../../components/Settings/Settings'

let changeLanguage; let showSettingsHandler; let selectedLanguage 
let showBackdrop; let hideBackdrop; let languageNames 
let theme; let themeListHandler; let text 
let unitListHandler; let units; let wrapper
beforeEach(() => {
  changeLanguage = jest.fn()
  showSettingsHandler = jest.fn()
  selectedLanguage = data.language
  showBackdrop = true
  hideBackdrop = jest.fn()
  languageNames = data.languageNames
  theme = data.theme
  themeListHandler = jest.fn()
  text = data.text.pl.settings
  unitListHandler = jest.fn()
  units = data.units.si
  wrapper = shallow(
    <Settings 
      showSettings={true}
      changeLanguage={changeLanguage}
      showSettingsHandler={showSettingsHandler}
      selectedLanguage={selectedLanguage}
      showBackdrop={showBackdrop}
      hideBackdrop={hideBackdrop}
      languageNames={languageNames}
      themeName={theme.name}
      themeListHandler={themeListHandler}
      text={text}
      unitListHandler={unitListHandler}
      units={units.id}
    />,
  )
})
// added Footer 
describe('Settings component tests', () => {
  test('Renders Setting components correctly', () => {
    
    expect(wrapper).toMatchSnapshot()
  })

  test('test language element render text for each language name and Icon component ', () => {
    const element = wrapper.find('.language label').at(0)
    expect(element.text()).toBe(`<Icon />${languageNames[0].name}`)
    
  })

  test('test language input funcionality ', () => {
    const payload = { value: languageNames[0].id } 
    const element = wrapper.find('.language input').at(0)
    element.simulate('change', payload)
    expect(changeLanguage).toHaveBeenLastCalledWith(payload)
    expect(element.props().value).toEqual(languageNames[0].id)
  })

  test('test theme element render text for each theme name and Icon component ', () => {
    const element = wrapper.find('.theme label').at(0)
    expect(element.text()).toBe(`<Icon />${theme.name}`)
  })

  test('test theme input functionality ', () => {
    const element = wrapper.find('.theme input').at(0)
    const payload = { value: theme.id } 
    element.simulate('change', payload)
    expect(themeListHandler).toHaveBeenLastCalledWith(payload)
    expect(element.props().value).toEqual(theme.id)
  })


  test('test unit element render text for each unit name and Icon component ', () => {
    const element = wrapper.find('.units label').at(0)
    expect(element.text()).toBe(`<Icon />${units.name}`)
    
  })

  test('test unit input functionality ', () => {
    const element = wrapper.find('.units input').at(0)
    const payload = { value: units.id } 
    element.simulate('change', payload)
    expect(unitListHandler).toHaveBeenLastCalledWith(payload)
    expect(element.props().value).toEqual(units.id)
  })
  
})