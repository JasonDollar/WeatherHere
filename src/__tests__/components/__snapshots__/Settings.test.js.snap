// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component should render correctly 1`] = `
<Fragment>
  <Backdrop
    hideBackdrop={[MockFunction]}
    showBackdrop={true}
  />
  <div
    className="Settings"
    id="Settings"
    style={
      Object {
        "opacity": "1",
        "transform": "translateY(0)",
      }
    }
  >
    <fieldset
      className="menu"
    >
      <legend>
        Zmień język
      </legend>
      <form
        className="form language"
      >
        <label
          className="label active"
          htmlFor="pl"
          key="pl"
        >
          <input
            checked={true}
            className="radio"
            id="pl"
            name="language"
            onChange={[MockFunction]}
            type="radio"
            value="pl"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          Polski
        </label>
      </form>
    </fieldset>
    <fieldset
      className="menu"
    >
      <legend>
        Zmień motyw
      </legend>
      <form
        className="form theme"
      >
        <label
          className="label"
          htmlFor="pink"
          key="Pink"
        >
          <input
            checked={false}
            className="radio"
            id="pink"
            name="theme"
            onChange={[MockFunction]}
            type="radio"
            value="pink"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          Pink
        </label>
        <label
          className="label"
          htmlFor="pinkDark"
          key="Pink Dark"
        >
          <input
            checked={false}
            className="radio"
            id="pinkDark"
            name="theme"
            onChange={[MockFunction]}
            type="radio"
            value="pinkDark"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          Pink Dark
        </label>
        <label
          className="label"
          htmlFor="pumpkin"
          key="Pumpkin Dark"
        >
          <input
            checked={false}
            className="radio"
            id="pumpkin"
            name="theme"
            onChange={[MockFunction]}
            type="radio"
            value="pumpkin"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          Pumpkin Dark
        </label>
        <label
          className="label"
          htmlFor="evening"
          key="Evening Dark"
        >
          <input
            checked={false}
            className="radio"
            id="evening"
            name="theme"
            onChange={[MockFunction]}
            type="radio"
            value="evening"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          Evening Dark
        </label>
        <label
          className="label"
          htmlFor="sky"
          key="Cool Sky"
        >
          <input
            checked={false}
            className="radio"
            id="sky"
            name="theme"
            onChange={[MockFunction]}
            type="radio"
            value="sky"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          Cool Sky
        </label>
        <label
          className="label"
          htmlFor="harvey"
          key="Harvey"
        >
          <input
            checked={false}
            className="radio"
            id="harvey"
            name="theme"
            onChange={[MockFunction]}
            type="radio"
            value="harvey"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          Harvey
        </label>
        <label
          className="label"
          htmlFor="spring"
          key="Spring"
        >
          <input
            checked={false}
            className="radio"
            id="spring"
            name="theme"
            onChange={[MockFunction]}
            type="radio"
            value="spring"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          Spring
        </label>
        <label
          className="label"
          htmlFor="lime"
          key="Lime Dark"
        >
          <input
            checked={false}
            className="radio"
            id="lime"
            name="theme"
            onChange={[MockFunction]}
            type="radio"
            value="lime"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          Lime Dark
        </label>
        <label
          className="label"
          htmlFor="vampire"
          key="Vampire Dark"
        >
          <input
            checked={false}
            className="radio"
            id="vampire"
            name="theme"
            onChange={[MockFunction]}
            type="radio"
            value="vampire"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          Vampire Dark
        </label>
      </form>
    </fieldset>
    <fieldset
      className="menu menuUnits"
    >
      <legend>
        Zmień jednostki
      </legend>
      <form
        className="form units"
      >
        <label
          className="label active"
          htmlFor="si"
          key="si"
        >
          <input
            checked={true}
            className="radio"
            id="si"
            name="units"
            onChange={[MockFunction]}
            type="radio"
            value="si"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          SI
        </label>
        <label
          className="label"
          htmlFor="us"
          key="us"
        >
          <input
            checked={false}
            className="radio"
            id="us"
            name="units"
            onChange={[MockFunction]}
            type="radio"
            value="us"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          US
        </label>
        <label
          className="label"
          htmlFor="uk2"
          key="uk2"
        >
          <input
            checked={false}
            className="radio"
            id="uk2"
            name="units"
            onChange={[MockFunction]}
            type="radio"
            value="uk2"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          UK
        </label>
        <label
          className="label"
          htmlFor="ca"
          key="ca"
        >
          <input
            checked={false}
            className="radio"
            id="ca"
            name="units"
            onChange={[MockFunction]}
            type="radio"
            value="ca"
          />
          <span
            className="radioButton"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Icon
              icon="checkmark"
              title=""
              viewBox="0 0 50 50"
            />
          </span>
          CA
        </label>
      </form>
    </fieldset>
    <Footer />
  </div>
</Fragment>
`;
